tests/test_tensorflow_data_api.py

```diff
@@ -28,253 +27,115 @@
-        tiledb_uri_y = os.path.join(tmpdir, "y" + array_uuid)
-
-        ingest_in_tiledb(
-            uri=tiledb_uri_x,
-            data=np.random.rand(ROWS, *input_shape[1:]),
-            batch_size=BATCH_SIZE,
-            sparse=False,
-            num_of_attributes=num_of_attributes,
-        )
-        ingest_in_tiledb(
-            uri=tiledb_uri_y,
-            data=np.random.rand(ROWS, NUM_OF_CLASSES),
-            batch_size=BATCH_SIZE,
-            sparse=False,
-            num_of_attributes=num_of_attributes,
-        )
-
-        with tiledb.open(tiledb_uri_x) as x, tiledb.open(tiledb_uri_y) as y:
-            tiledb_dataset = TensorflowTileDBDataset(
-                x_array=x,
-                y_array=y,
-                batch_size=BATCH_SIZE,
-                batch_shuffle=batch_shuffle,
-                buffer_size=buffer_size,
-                within_batch_shuffle=within_batch_shuffle,
-                x_attrs=["features_" + str(attr) for attr in range(num_of_attributes)],
-                y_attrs=["features_" + str(attr) for attr in range(num_of_attributes)],
-            )
-
-            assert isinstance(tiledb_dataset, tf.data.Dataset)
-
-            # Same test without attribute names explicitly provided by the user
-            tiledb_dataset = TensorflowTileDBDataset(
-                x_array=x,
-                y_array=y,
-                batch_size=BATCH_SIZE,
-                batch_shuffle=batch_shuffle,
-                buffer_size=buffer_size,
-                within_batch_shuffle=within_batch_shuffle,
-            )
-
-            assert isinstance(tiledb_dataset, tf.data.Dataset)
-
-    def test_except_with_diff_number_of_x_y_rows(
+@pytest.mark.parametrize("input_shape", [(10,), (10, 3)])
+@pytest.mark.parametrize("num_attrs", [1, 2])
+@pytest.mark.parametrize("batch_shuffle", [True, False])
+@pytest.mark.parametrize("within_batch_shuffle", [True, False])
+@pytest.mark.parametrize("buffer_size", [50, None])
+class TestTensorflowTileDBDatasetDense:
+    @pytest.mark.parametrize("sparse_y", [True, False])
+    def test_dense_x(
         self,
         tmpdir,
         input_shape,
-        num_of_attributes,
+        num_attrs,
         batch_shuffle,
         within_batch_shuffle,
         buffer_size,
+        sparse_y,
     ):
-        array_uuid = str(uuid.uuid4())
-        tiledb_uri_x = os.path.join(tmpdir, "x" + array_uuid)
-        tiledb_uri_y = os.path.join(tmpdir, "y" + array_uuid)
-
-        ingest_in_tiledb(
-            uri=tiledb_uri_x,
-            # Add one extra row on X
-            data=np.random.rand(ROWS + 1, *input_shape[1:]),
+        if sparse_y and within_batch_shuffle:
+            pytest.skip("Sparse y not supported with within_batch_shuffle")
+
+        uri_x, uri_y = ingest_in_tiledb(
+            tmpdir,
+            data_x=np.random.rand(ROWS, *input_shape),
+            data_y=create_rand_labels(ROWS, NUM_OF_CLASSES, one_hot=sparse_y),
```

Comment:
@gsakkis do you want to leave it as is, i.e., NUM_OF_CLASSES, which is a little bit misleading as we discussed, or give a more generic name, e.g., y_dim or something similar?